### PR TITLE
bbatha/pla 1428 setup docker registry mirror in cluster

### PR DESCRIPTION
**Pull request by bbatha**
Graphcore needs this now. We will roll out to more clusters in the future.

### CODE REVIEW

File: gradient-metal-gc/main.tf [hunk 1]
```diff
@@ -123,3 +118,16 @@ module "gradient_processing" {
   notebook_volume_type                                = var.notebook_volume_type
 }
 
+
+module "container_registry_mirror" {
```
Comment (brodeyn) [inline gradient-metal-gc/main.tf:122]: we only care to deploy this with gc?

File: modules/container-registry-mirror/files/docker-registry-mirror.yaml.tpl [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    bucket: "${s3.bucket}"
+%{ endif }
+
+proxy:
+    username: "${docker_username}"
+    password: "${docker_password}"
+
+haSharedSecret: "${ha_shared_secret}"
+
+%{ if s3 != null }
```
Comment (brodeyn) [inline modules/container-registry-mirror/files/docker-registry-mirror.yaml.tpl:21]: can add this to the `%{ if s3 != null }` above

File: modules/container-registry-mirror/main.tf [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    name = "docker-registry"
+  }
+  lifecycle {
+    ignore_changes = [metadata]
+  }
+}
+
+resource "helm_release" "docker_mirror" {
+  name       = "docker-registry-mirror"
+  version    = "1.10.3"
```
Comment (brodeyn) [inline modules/container-registry-mirror/main.tf:18]: maybe make this version configurable? and default it to `"1.10.3"`

File: gradient-metal-gc/main.tf (hunk 1 above)
Comment (bbatha) [reply gradient-metal-gc/main.tf:122]: For the moment, we'll be trying to do it for the public cluster in an upcoming pr

**Review by brodeyn — APPROVED** [review]

**Pull request merged**